tests/functional/avatar.js

```diff
@@ -160,122 +160,130 @@ define([
-            .end()
+        // accept permission
+        .findById('accept')
+          .click()
+        .end()
 
-            .findByCssSelector('a.cancel')
-              .click()
-            .end()
+        .findByCssSelector('img[src*="https://secure.gravatar.com"]')
+        .end()
 
-            // redirected back to main avatar page after save
-            .findById('avatar-options')
-            .end()
+        .findByCssSelector('a.cancel')
+          .click()
+        .end()
 
-            // give time for error to show up, there should be no error though
-            .sleep(500)
+        // redirected back to main avatar page after save
+        .findById('avatar-options')
+        .end()
 
-            .findByCssSelector('.error')
-              .getVisibleText()
-              .then(function (val) {
-                assert.ok(! val, 'has no error text');
-              })
-            .end()
+        // give time for error to show up, there should be no error though
+        .sleep(500)
 
-            // success is seeing no profile image set
-            .waitForDeletedByCssSelector('.avatar-wrapper img')
-            .end();
-        });
+        .findByCssSelector('.error')
+          .getVisibleText()
+          .then(function (val) {
+            assert.ok(! val, 'has no error text');
+          })
+        .end()
+
+        // success is seeing no profile image set
+        .waitForDeletedByCssSelector('.avatar-wrapper img')
+        .end();
     },
 
     'visit gravatar with no gravatar set': function () {
       var self = this;
-      return signUp(self, emailAvatarAb)
-        .then(function () {
-          return self.remote
-            .get(require.toUrl(AVATAR_CHANGE_URL))
+      return self.remote
+        .get(require.toUrl(AVATAR_CHANGE_URL))
 
-            // go to change avatar
-            .findById('gravatar')
-              .click()
-            .end()
+        // go to change avatar
+        .findById('gravatar')
+          .click()
+        .end()
 
-            // success is going to the change avatar page
-            .findById('avatar-options')
-            .end()
+        // accept permission
+        .findById('accept')
+          .click()
+        .end()
 
-            // success is seeing the error text
-            .then(FunctionalHelpers.visibleByQSA('.error'))
-            .findByCssSelector('.error')
-              .getVisibleText()
-              .then(function (val) {
-                assert.ok(val, 'has error text');
-              })
-            .end();
-        });
+        // success is going to the change avatar page
+        .findById('avatar-options')
+        .end()
+
+        // success is seeing the error text
+        .then(FunctionalHelpers.visibleByQSA('.error'))
+        .findByCssSelector('.error')
+          .getVisibleText()
+          .then(function (val) {
+            assert.ok(val, 'has error text');
+          })
+        .end();
```

Comment:
It looks like a test is needed for when the user has already granted permissions to ensure they do not have to re-grant permissions.